cargo-shuttle/src/lib.rs

```diff
@@ -901,15 +901,45 @@ impl Shuttle {
     }
 
     async fn deploy(&self, client: &Client, args: DeployArgs) -> Result<CommandOutcome> {
-        if !args.allow_dirty {
-            self.is_dirty()?;
+        let working_directory = self.ctx.working_directory();
+
+        let mut deploy_req: DeploymentRequest = DeploymentRequest {
+            no_test: args.no_test,
+            ..Default::default()
+        };
+
+        if let Ok(repo) = Repository::discover(working_directory) {
+            let repo_path = repo
+                .workdir()
+                .context("getting working directory of repository")?;
+            let repo_path = dunce::canonicalize(repo_path)?;
+            trace!(?repo_path, "found git repository");
+
+            let is_dirty = self.is_dirty(&repo).err();
+            if !args.allow_dirty {
+                if let Some(err) = is_dirty {
+                    bail!(err);
+                }
+            }
+            deploy_req.git_dirty = Some(is_dirty.is_some());
+
+            if let Ok(head) = repo.head() {
+                // This is typically the name of the current branch
+                // It is "HEAD" when head detached, for example when a tag is checked out
+                deploy_req.git_branch = head.shorthand().map(String::from);
+                if let Ok(commit) = head.peel_to_commit() {
+                    let commit_id = commit.id();
+                    deploy_req.git_commit_id = Some(commit_id.to_string());
+                    // message is valid utf-8
+                    deploy_req.git_commit_msg = commit.summary().map(|s| s.chars().take(80).collect());
```

Comment:
I implemented string trimming on both ends like this.
Ideally it would trim all incoming strings that are going to be put into a DB.
What do you think? (will use shared constant if approved)

Reply:
Looking good.

Reply:
sgtm